assets/arc-0060/arc60wallet.api.ts

```diff
@@ -0,0 +1,324 @@
+     * 
+     */
+    constructor(private readonly k: Uint8Array) {
+        this.k = k;
+    }
+
+    /**
+     * Arbitrary data signing function. 
+     * Based on the provided scope and encoding, it decodes the data and signs it.
+     * 
+     * 
+     * @param signingData - includes the data to be signed and the signer's public key
+     * @param metadata - includes the scope and encoding of the data
+     * @returns - signature of the data from the signer.
+     * 
+     * @throws - Error 4600 - Invalid Scope - if the scope is not supported
+     * @throws - Error 4601 - DOesn't match schema - The data doesn't match the schema for the scope
+     * @throws - Error 4602 - Failed decoding - if the data can't be decoded
+     * @throws - Error 4603 - Invalid Signer - if the signer is not a valid public key
+     * @throws - Error 4604 - Invalid HD Path - if the HD path is invalid
+     */
+    async signData(signingData: StdSigData, metadata: StdSignMetadata): Promise<StdSignature> {
+        // decode signing data with chosen metadata.encoding
+        let decodedData: Uint8Array
+        let toSign: Uint8Array
+
+        // decode data
+        switch(metadata.encoding) {
+            case 'base64':
+                decodedData = Buffer.from(signingData.data, 'base64');
+                break;
+            case 'hex':
+                decodedData = Buffer.from(signingData.data, 'hex');
+                break;
+            default:
+                throw ERROR_FAILED_DECODING;
+        }
+
+        // Reject if Program bytes prefix is present in byte array
+        if(decodedData.slice(0, 7).toString() === "Program") {
+            throw ERROR_DOESNT_MATCH_SCHEMA;
+        }
+        
+        // validate against schema
+        switch(metadata.scope) {
+            case ScopeType.CHALLENGE32:
+                if(decodedData.length !== 32) {
+                    throw ERROR_DOESNT_MATCH_SCHEMA;
+                }
+
+                toSign = decodedData;
+                break;
+            case ScopeType.MX_RANDOM:
+                // Check MX prefix
+                if(decodedData[0] !== 0x6D || decodedData[1] !== 0x78) {
+                    throw ERROR_DOESNT_MATCH_SCHEMA;
+                }
+
+                toSign = decodedData
+                break;
+            case ScopeType.LSIG_TEMPLATE:
```

Comment:
The `LSIG_TEMPLATE` thing is strange/no-user-friendly. IMO, we either allow the signing of LSIGS (users are responsible for passing the correct data with a schema) or remove this scope.

Reply:
Disagree on this. If we use templates and integrity checks for known templates; this API becomes safe as no unknown LSIGS can be signed. Just allowing LSIGS will open a can of worms and nightmare for apps / wallets to offer safe. 

I see very little to no risk in doing so by using templates